app/src/main/java/com/example/pinkorange/vibrato/MainActivity.java

```diff
@@ -1,35 +1,62 @@
 package com.example.pinkorange.vibrato;
 
+import android.Manifest;
 import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.util.Log;
 import android.view.View;
 import android.widget.Button;
 
 public class MainActivity extends AppCompatActivity {
-    protected Intent recordedIntent;
+    protected Intent recordedIntent, liveIntent;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        requestPermissions();
+
         Button recorded_music_button = findViewById(R.id.recorded_music_button);
         recordedIntent = new Intent(MainActivity.this, SelectMusic.class);
         recorded_music_button.setOnClickListener(new View.OnClickListener() {
              public void onClick(View v) {
-                 Intent intent = new Intent(MainActivity.this, SelectMusic.class);
-                 startActivity(intent);
+                 startActivity(recordedIntent);
+             }
+         });
+
+        Button live_music_button = findViewById(R.id.live_music_button);
+        liveIntent = new Intent(MainActivity.this, LiveWithSettings.class);
+        live_music_button.setOnClickListener(new View.OnClickListener() {
+             public void onClick(View v) {
+                 Log.e("App", "ASIOHXIOH");
+                 liveIntent.putExtra("live", true);
+                 startActivity(liveIntent);
              }
          });
+    }
 
-        /*Button bass = findViewById(R.id.buttonBass);
-        bass.setOnClickListener(new View.OnClickListener() {
-            public void onClick(View v)
-            {
-                Intent i = new Intent(MainActivity.this, BassBooster.class);
-                startActivity(i);
-            }
-        });*/
+    private void requestPermissions() {
```

Comment:
Would it be possible to move this function into a permission utils file or something, and modify it to have a `applicationContext` parameter so we have flexibility in the future on where to call it from + can maybe call it from multiple different places?